### ISSUE TITLE
Add `.finally`

### ISSUE THREAD

**Issue by trufae** [tj/co#315]
Bluebird already provides the `.finally` logic which in some situations makes the code cleaner. it will be good if `co` can have it too.

**Comment by kyeotic** [comment]
Javascript already has a [finally](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#The_finally_clause). Just use it

```javascript
co(function * () {
  try {
   // try to work
  } catch (e) {
  // handle your errors
  } finally {
    // cleanup
  }
})
```

**Comment by luizamboni** [comment]
> Javascript already has a finally. Just use it

Yes, all of us, know this, but the intend to use a module like co is invert this control...
to can use a finally in a generator without build a other generator method to wrap try, catch and finally

The same logic can be applied from catch too.. "javascript alreay has a catch" then why co need have one ?
Make sense ?


example:

```
const obj = {
  insertInDb: function*(attrs){
     //without try ...
  }
}

co(obj.insertInDb(attrs)).catch(...).finally(..close connections etc)

```

**Comment by kyeotic** [comment]
Co doesn't add a catch either, because you don't need one. You use the js keywords

```javascript
const obj = {
  insertInDb: function*(attrs){
     //without try ...
  }
}

co(function * () {
  try {
    yield obj.insertInDb(attrs)
  } catch e() {
    ...
  } finally{ 
    ..close connections etc
  }
})

**Comment by kyeotic** [comment]
The `finally` in this line

```
co(obj.insertInDb(attrs)).catch(...).finally(..close connections etc)
```

Would have to be on the `Promise.prototype`. `co` returns Promises.

**Comment by vedmant** [comment]
You can use .then() after .catch() it will work as .finally() https://stackoverflow.com/a/35999141/1753349
